pkg/pillar/cmd/domainmgr/domainmgr.go

```diff
@@ -1674,7 +1675,23 @@ func doActivateTail(ctx *domainContext, status *types.DomainStatus,
 	status.VifList = checkIfEmu(status.VifList)
 
 	status.State = types.RUNNING
-	domainID, state, err := hyper.Task(status).Info(status.DomainName)
+	var state types.SwState
+	// If status is unknown, get info again for some retries and if it will not be activated return error.
+	for retry := 0; retry < unknownStateRetries; retry++ {
+		ctx.ps.StillRunning(agentName, warningTime, errorTime)
+		domainID, state, err = hyper.Task(status).Info(status.DomainName)
+		if err != nil || state != types.UNKNOWN {
+			break
+		}
+		log.Warnf("doActivateTail(%v) for %s: state is UNKNOWN, retry %d", status.UUIDandVersion, status.DisplayName, retry)
```

Comment:
```suggestion
		log.Warnf("doActivateTail(%v) for %s: state is UNKNOWN, retry %d/%d", status.UUIDandVersion, status.DisplayName, retry, unknownStateRetries)
```
just a small thing - I think it's better readable this way
(sorry, I was too late on the main PR)

Reply:
I would prefer not to fix it in a backport PR. But we can improve it in master, the idea is nice!